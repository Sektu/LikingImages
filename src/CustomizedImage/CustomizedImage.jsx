import LikeButton from "../LikeButton/LikeButton";
import DislikeButton from "../DislikeButton/DislikeButton";

const CustomizedImage = ({name, src, fullName}) => {
    return (
        <>
            <h5 className="image-name">{fullName}</h5>
            <img className="image" src={src} alt={name} />
            <LikeButton onClick={() => {}} />
            <DislikeButton onClick={() => {}} />
        </>
    );
}

export default CustomizedImage;