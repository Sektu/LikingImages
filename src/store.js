import { configureStore } from "@reduxjs/toolkit";
import likingReducer from "./slices/likingSlice";
import imagesReducer from "./slices/imagesSlice";

export default configureStore({
  reducer: {
    liking: likingReducer,
    images: imagesReducer,
  }
});
