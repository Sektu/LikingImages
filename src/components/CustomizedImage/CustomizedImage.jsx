import { useSelector, useDispatch } from "react-redux";
import { doLike, doDislike } from "../../slices/likingSlice";
import LikeButton from "../LikeButton/LikeButton";
import DislikeButton from "../DislikeButton/DislikeButton";
import "./CustomizedImage.css";

const CustomizedImage = ({name, src, fullName}) => {
    const likingData = useSelector((state) => state.liking.data);
    const dispatch = useDispatch();
    const isLiked = likingData[name];
    return (
        <>
            <h5 className="image-name">{fullName}</h5>
            <img className="image" src={src} alt={name} />
            <div>
            {
                isLiked ? 
                    <DislikeButton onClick={() => {dispatch(doDislike(name))}} />
                    : 
                    <LikeButton onClick={() => {dispatch(doLike(name))}} />
            }
            </div>
        </>
    );
}

export default CustomizedImage;