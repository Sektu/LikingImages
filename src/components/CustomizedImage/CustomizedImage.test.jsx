import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CustomizedImage from './CustomizedImage';
import store from "../../store";
import { Provider } from "react-redux";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<Provider store={store}><CustomizedImage /></Provider>);
  expect(asFragment()).toMatchSnapshot();
});

it("check image name", () => {
    const name = "batman";
    const src = "https://fwcdn.pl/fpo/63/18/626318/7998475.3.jpg";
    const { getByText } = render(<Provider store={store}><CustomizedImage name={name} src={src} /></Provider>);

    const imageName = getByText("batman");
    expect(imageName).toBeInTheDocument();
    expect(imageName).toBeVisible();
    expect(imageName).toHaveClass('image-name');
    expect(imageName).not.toHaveClass('extra');
});

it("check image", () => {
    const name = "batman";
    const src = "https://fwcdn.pl/fpo/63/18/626318/7998475.3.jpg";
    const { getByAltText } = render(<Provider store={store}><CustomizedImage name={name} src={src} /></Provider>);

    const image = getByAltText("batman");
    expect(image).toBeInTheDocument();
    expect(image).toBeVisible();
    expect(image).toHaveClass('image');
    expect(image).toHaveAttribute('src', src);
    expect(image).toHaveAttribute('alt', name);
});

it("check like/dislike button", () => {
    const name = "batman";
    const src = "https://fwcdn.pl/fpo/63/18/626318/7998475.3.jpg";
    const { getByRole } = render(<Provider store={store}><CustomizedImage name={name} src={src} /></Provider>);

    const button = getByRole("button");
    expect(button).toBeInTheDocument();
    expect(button).toBeVisible();
    expect(button).toHaveClass('button');
    expect(button).toHaveClass('like-button');
});