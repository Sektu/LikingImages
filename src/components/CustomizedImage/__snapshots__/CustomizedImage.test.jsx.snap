// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <h5
    class="image-name"
  />
  <img
    class="image"
  />
  <div>
    <button
      class="button like-button"
    >
      Like
    </button>
  </div>
</DocumentFragment>
`;
